src/BuildToolchains.jl

```diff
@@ -0,0 +1,203 @@
+abstract type AbstractBuildToolchain{C} end
+
+struct CMake{C} <: AbstractBuildToolchain{C} end
+struct Meson{C} <: AbstractBuildToolchain{C} end
+
+c_compiler(::AbstractBuildToolchain{:clang}) = "clang"
+cxx_compiler(::AbstractBuildToolchain{:clang}) = "clang++"
+c_compiler(::AbstractBuildToolchain{:gcc}) = "gcc"
+cxx_compiler(::AbstractBuildToolchain{:gcc}) = "g++"
+
+function cmake_arch(p::AbstractPlatform)
+    if arch(p) == "powerpc64le"
+        return "ppc64le"
+    else
+        return arch(p)
+    end
+end
+
+function cmake_os(p::AbstractPlatform)
+    if Sys.islinux(p)
+        return "Linux"
+    elseif Sys.isfreebsd(p)
+        return "FreeBSD"
+    elseif Sys.isapple(p)
+        return "Darwin"
+    elseif Sys.iswindows(p)
+        return "Windows"
+    end
+end
+
+function toolchain_file(bt::CMake, p::AbstractPlatform)
+    target = triplet(p)
+    aatarget = aatriplet(p)
+
+    if Sys.isapple(p)
+        return """
+        # CMake toolchain file for $(c_compiler(bt)) running on $(target)
+        set(CMAKE_SYSTEM_NAME $(cmake_os(p)))
+        set(CMAKE_SYSTEM_PROCESSOR $(cmake_arch(p)))
+        set(CMAKE_SYSTEM_VERSION 14.5)
```

Comment:
I think this is the version of Darwin?  Maybe we can use `os_version(p)` instead of hard-coding 14.5?

Reply:
Agreed.  Let's use `os_version(p)`.